components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java

```diff
@@ -71,4 +85,44 @@ public Response systemScopesGet(MessageContext messageContext) throws APIManagem
         }
         return null;
     }
+
+    public Response updateRolesForScope(ScopeListDTO body, MessageContext messageContext)
+            throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(RestApiUtil.getLoggedInUsername());
+        //read from tenant-conf.json
+        JsonObject existingTenantConfObject = new JsonObject();
+        try {
+            APIMRegistryService apimRegistryService = new APIMRegistryServiceImpl();
+            String existingTenantConf = apimRegistryService.getConfigRegistryResourceContent(tenantDomain,
+                    APIConstants.API_TENANT_CONF_LOCATION);
+            existingTenantConfObject = new JsonParser().parse(existingTenantConf).getAsJsonObject();
+        } catch (RegistryException | UserStoreException e) {
+            APIUtil.handleException("Couldn't read tenant configuration from tenant registry", e);
+        }
+        JSONObject responseJson = SystemScopesMappingUtil.createJsonObjectOfScopeMapping(body);
+        //Here we are removing RESTAPIScopes from the existing tenant-conf
+        // Adding new RESTAPIScopes to the existing tenant-conf.
+        existingTenantConfObject.remove(APIConstants.REST_API_SCOPES_CONFIG);
+        JsonElement jsonElement = new JsonParser().parse(responseJson.toJSONString());
+        existingTenantConfObject.add(APIConstants.REST_API_SCOPES_CONFIG, jsonElement);
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            String formattedTenantConf = mapper.writerWithDefaultPrettyPrinter()
+                    .writeValueAsString(existingTenantConfObject.toString());
+            APIUtil.updateTenantConf(existingTenantConfObject.toString(), tenantDomain);
+            if (log.isDebugEnabled()) {
+                log.debug("Finalized tenant-conf.json: " + formattedTenantConf);
+            }
+        } catch (JsonProcessingException e) {
+            throw new APIManagementException("Error while formatting tenant-conf.json of tenant ");
```

Comment:
Let's not omit 'e' please. Shall we include it in the throwed APIManagementException?